query.go

```diff
@@ -101,11 +117,19 @@ func NewQuery(name string, qtype uint16, options ...QueryOption) (*dns.Msg, erro
 		Qclass: dns.ClassINET,
 	}
 	query := new(dns.Msg)
-	query.Id = dns.Id()
 	query.RecursionDesired = true
 	query.Question = make([]dns.Question, 1)
 	query.Question[0] = question
 
+	// Only set the queryID for protocols that actually
+	// require a nonzero queryID to be set.
+	switch serverAddr.Protocol {
+	case ProtocolDoH:
+		// the zero ID MUST be used
```

Comment:
For DoH and DoQ cases, we can write `query.id = 0` right?

Reply:
Correct! I felt it was OK to do nothing since the `query` is zero-initialised by `new(dns.Msg)`. I will slightly update the comment to acknowledge that the ID is already zero, to make the comment less ambiguous.